Tools/CISupport/ews-build/steps.py

```diff
@@ -959,54 +1051,104 @@ def skip_build(self, reason):
         bug_id = self.getProperty('bug_id', '') or self.get_bug_id_from_patch(patch_id)
 
         bug_closed = self._is_bug_closed(bug_id) if self.verifyBugClosed else 0
         if bug_closed == 1:
             self.skip_build('Bug {} is already closed'.format(bug_id))
-            return None
+            return False
 
         obsolete = self._is_patch_obsolete(patch_id) if self.verifyObsolete else 0
         if obsolete == 1:
             self.skip_build('Patch {} is obsolete'.format(patch_id))
-            return None
+            return False
 
         review_denied = self._is_patch_review_denied(patch_id) if self.verifyReviewDenied else 0
         if review_denied == 1:
             self.skip_build('Patch {} is marked r-'.format(patch_id))
-            return None
+            return False
 
         cq_plus = self._is_patch_cq_plus(patch_id) if self.verifycqplus else 1
         if cq_plus != 1:
             self.skip_build('Patch {} is not marked cq+.'.format(patch_id))
-            return None
+            return False
 
         acceptable_review_flag = self._does_patch_have_acceptable_review_flag(patch_id) if self.verifycqplus else 1
         if acceptable_review_flag != 1:
             self.skip_build('Patch {} does not have acceptable review flag.'.format(patch_id))
-            return None
+            return False
 
         if obsolete == -1 or review_denied == -1 or bug_closed == -1:
             self.finished(WARNINGS)
-            return None
+            return False
+        return True
 
-        if self.verifyBugClosed:
-            self._addToLog('stdio', 'Bug is open.\n')
-        if self.verifyObsolete:
-            self._addToLog('stdio', 'Patch is not obsolete.\n')
-        if self.verifyReviewDenied:
-            self._addToLog('stdio', 'Patch is not marked r-.\n')
-        if self.verifycqplus:
-            self._addToLog('stdio', 'Patch is marked cq+.\n')
-            self._addToLog('stdio', 'Patch have acceptable review flag.\n')
-        self.finished(SUCCESS)
-        return None
+    def validate_github(self, pr_number):
+        if not pr_number:
+            return False
+
+        repository_url = self.getProperty('repository', '')
+        title = self.getProperty('github.title', '')
+        owners = self.getProperty('owners', [])
+
+        if self.addURLs and pr_number and title and repository_url:
+            self.addURL('[PR {}] {}'.format(pr_number, title), GitHub.pr_url(pr_number, repository_url))
+        if self.addURLs and owners:
+            contributors, errors = Contributors.load(use_network=False)
+            for error in errors:
+                print(error)
+                self._addToLog('stdio', error)
+            contributor = contributors.get(owners[0], {})
+            display_name = contributor.get('email', owners[0]) if contributor else owners[0]
+            self.addURL('Change by {}'.format(display_name), '{}/{}'.format(GITHUB_URL, owners[0]))
```

Comment:
I'm still unsure if we should do this contributors lookup and try to display the full name instead of github username. Especially because there might be lot of people creating PRs who wouldn't be in contributors.json (currently we do have many people uploading patches who are not in contributors.json). For them we will fail to display the full name and display the github username instead. 

Effectively, this will cause inconsistency in the display. For some people we display full name, and for some we display github username. Might as well just display github username for everyone (or maybe atleast for first iteration). We might want to get feedback from our users on it though.

Reply:
My concern is specifically about long-time contributors. Many of their GitHub accounts pre-date their involvement with the WebKit project and don't align with their Slack usernames or email addresses. As an example, Zhifei Fang is @facetothefate, which is something I expect will cause confusion for folks (it already has for me, and is a big reason I asked folks to start adding their GitHub usernames to contributors.json in the first place)